core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -432,7 +500,7 @@ private void deleteRemovedMetadataFiles(TableMetadata base, TableMetadata metada
     if (deleteAfterCommit) {
       Set<TableMetadata.MetadataLogEntry> removedPreviousMetadataFiles =
           Sets.newHashSet(base.previousFiles());
-      removedPreviousMetadataFiles.removeAll(metadata.previousFiles());
```

Comment:
Why is this changed

Reply:
This is an optimisation introduced by the IDE, where there was an efficiency issue with the execution of the removeAll method. I don't think this change is important, so I haven't reverted it.